### PR TITLE
Fix certificates handling

### PR DESCRIPTION
**Pull request by imobachgs**
This PR should fix some certificates issues during installation:

* [bsc#981428](https://bugzilla.suse.com/show_bug.cgi?id=981428)
* [bsc#989787](https://bugzilla.suse.com/show_bug.cgi?id=989787)
* [bsc#991028](https://bugzilla.suse.com/show_bug.cgi?id=991028)

### CODE REVIEW

**Comment by kobliha** [comment]
`Coverage decreased (-72.4%) to 12.152%` -- this must be some mistake in Coveralls?

**Comment by imobachgs** [comment]
@kobliha Unlike other YaST2 projects, this one has a `Gemfile` and I forgot to add the cheetah gem. Fixed now: `Coverage increased (+0.2%)` :)

File: src/lib/registration/ssl_certificate.rb [hunk 1]
```diff
@@ -3,13 +3,35 @@
...
     attr_reader :x509_cert
 
+    # Path to store the certificate of the registration server
+    #
+    # During installation, the certificate should be written to a read-write
+    # directory. On an installed system, the method relies in SUSEConnect.
+    #
+    # @return [String] Path to store the certificate
+    def self.default_certificate_path
+      if Yast::Mode.installation || Yast::Mode.update
```
Comment (jreidinger) [inline src/lib/registration/ssl_certificate.rb]: why not `Stage.initial`?

Comment (imobachgs) [inline src/lib/registration/ssl_certificate.rb]: Yes, you're right. It always happens during 1st stage.

File: src/lib/registration/ssl_certificate.rb [hunk 2]
```diff
@@ -28,6 +50,42 @@ def self.download(url, insecure: false)
...
+        files.each do |file|
+          # FileUtils.cp does not seem to allow copying the links without dereferencing them.
+          Yast::Execute.locally("cp", "--no-dereference", "--preserve=links", file, subdir)
+        end
+      end
+
+      # Cleanup
+      FileUtils.rm_rf(TMP_CA_CERTS_DIR)
+      true
+    rescue Cheetah::ExecutionFailed => e
```
Comment (jreidinger) [inline src/lib/registration/ssl_certificate.rb]: cannot happen, see how Execute is implemented with catching exception

File: src/lib/registration/ssl_certificate.rb [hunk 3]
```diff
@@ -84,8 +142,48 @@ def fingerprint(sum)
...
+    #
+    # @return [true] true if import was successful
+    #
+    # @raise Connect::SystemCallError
+    # @raise Cheetah::ExecutionFailed
+
+    # @see #import_to_system
+    # @see #import_to_instsys
+    def import
+      (Yast::Mode.installation || Yast::Mode.update) ? import_to_instsys : import_to_system
```
Comment (jreidinger) [inline src/lib/registration/ssl_certificate.rb]: same here. for insts-sys detection `Stage.initial` is better

File: src/lib/registration/ssl_certificate.rb (hunk 2 above)
Comment (imobachgs) [inline src/lib/registration/ssl_certificate.rb]: Then I guess that [this documentation](https://github.com/yast/yast-yast2/blob/master/library/system/src/lib/yast2/execute.rb#L56) is outdated, right?

File: src/lib/registration/ssl_certificate.rb [hunk 4]
```diff
@@ -69,21 +65,23 @@ def self.update_instsys_ca
 
       # Copy certificates/links
       files = Dir[File.join(TMP_CA_CERTS_DIR, "*")]
+      return false if files.empty?
       targets = ["pem", "openssl"].map { |d| File.join(CA_CERTS_DIR, d) }
-      targets.each do |subdir|
+      new_files = []
+      targets.each_with_object(new_files) do |subdir|
```
Comment (jreidinger) [inline src/lib/registration/ssl_certificate.rb]: it should be `new_files = targets.each_with_object([]) do |subdir, new_files|`

**Comment by jreidinger** [comment]
one note and travis failed, otherwise looks good

**Comment by jreidinger** [comment]
LGTM if travis is happy

**Pull request merged**